Developer pain

Potherca opened the pull request:
Besides adding prose explaing Developer Pain and CSS/JS which adds a Developer Pain Calculator to the Developer Pain Criteria page, this PR also adds:

- A linter for Liquid template files
- A Yarn lock file
- Some quotes on the topic of pain

The developer pain calculator can be seen working here: https://potherca-contrib.github.io/guides/protocols/estimating/developer-pain/criteria/

Potherca committed:
CHANGE: Adds some quotes on the topic of pain.

Potherca committed:
CHANGE: Adds liquid linter to `npm test` command.

Potherca committed:
FIX: Liquid linter fixes.

Potherca committed:
CHANGE:Whitespace fixes for definition of blocking README.md.

Potherca committed:
ADD: Adds yarn.lock file.

Potherca committed:
CHANGE: Fixes typo.

Potherca committed:
ADD: Adds the ability to add page specific javascript.

Potherca committed:
ADD: Adds Octicons from CDN.

Potherca committed:
ADD: Adds developer pain criteria.

Potherca committed:
ADD: Adds explanation for developer pain.

Potherca committed:
ADD: Adds developer pain to protocol index.

Potherca committed:
ADD: Adds styling for developer pain calculator.

Potherca committed:
ADD: Adds javascript logic for developer pain calculator.

Potherca committed:
FIX: Whitespace fix.

Potherca committed:
FIX: Fixes two bugs in the page specific JS logic

 - Makes sure previously defined tags are not (re)used, this lead to CSS being included in the page twice.
 - Adds the base URL to the JS path as this causes problem when the guides repo lives in a sub-folder (rather than a sub-domain).

Marascon commented on _includes/page-specific-js.lqd:
```diff
@@ -0,0 +1,28 @@
+        {% for static_file in site.static_files %}
+            {% if static_file.path == filePath %}
+                {% capture output %}
+  <script src="{{ filePath | prepend:site.baseurl }}"></script>{% endcapture  %}
```
Indenting seems a little off

Potherca replied:
Indenting is as intended. In order to avoid extra indenting where this macro is used, no whitespace is added here.

Marascon reviewed (commented):
Might want to check that indent

Marascon reviewed (approved):
Seems all good to me, also no flaws in the documentation from how i read it.

Potherca committed:
CHANGE: Adds CSS helper class to hide elements.

Potherca committed:
CHANGE: Makes protocols-estimating-developer-pain-criteria.css more BEM compliant.

Potherca committed:
CHANGE: Cleanup for protocols-estimating-developer-pain-criteria.js file

 - Places some logic in functions
 - Adds class to help icon
 - Adds class to hide <sup>
